parlai/agents/tfidf_retriever/build_db.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+"""A script to read in and store ParlAI tasks in a sqlite database.
+
+Adapted from Adam Fisch's work at github.com/facebookresearch/DrQA/
+"""
+
+import argparse
+import sqlite3
+import json
+import os
+import logging
+import importlib.util
+
+from multiprocessing import Pool as ProcessPool
+from tqdm import tqdm
+from . import utils
+
+from collections import deque
+import random
+from parlai.core.agents import create_task_agent_from_taskname
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+fmt = logging.Formatter('%(asctime)s: [ %(message)s ]', '%m/%d/%Y %I:%M:%S %p')
+console = logging.StreamHandler()
+console.setFormatter(fmt)
+logger.addHandler(console)
+
+# ------------------------------------------------------------------------------
+# Store corpus.
+# ------------------------------------------------------------------------------
+
+def store_contents(opt, task, save_path, context_length=-1, include_labels=True):
```

Comment:
as i said before,  i thought it might be better to not do it this but to use the standard act() function like we do to build a dictionary, but maybe i'm wrong maybe this is simpler.
i think a workaround would be to allow both ways to do it? can that be done?

Reply:
if you don't specify a task, it doesn't do this--and it does allow adding them via the act function as well (that is, both ways are allowed!)